src/H5Dchunk.c

```diff
@@ -961,6 +961,13 @@ H5D__chunk_init(H5F_t *f, const H5D_t *const dset, hid_t dapl_id)
         HGOTO_ERROR(H5E_DATASET, H5E_CANTINIT, FAIL, "unable to set # of chunks for dataset");
 
 done:
+    if (FAIL == ret_value) {
+        if (rdcc->slot)
+            rdcc->slot = H5FL_SEQ_FREE(H5D_rdcc_ent_ptr_t, rdcc->slot);
+
+        if (sc->ops->dest && (sc->ops->dest)(&idx_info) < 0)
```

Comment:
I believe we need to keep track of whether the init function was called. Using b-tree v1 as an example the dest callback appears to always call H5UC_DEC on the shared data, and H5UC_DEC assumes the shared data exists.

Reply:
Okay, I'll add a flag when I get back from lunch.